lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -1712,11 +1763,41 @@ AddressMaterialization::materializeProjectionIntoUse(Operand *operand,
       return pass.function->getArguments()[resultIdx];
     }
     SILValue tupleAddr = materializeComposingUser(tupleInst, intoPhiOperand);
-    return B.createTupleElementAddr(pass.genLoc(), tupleAddr,
-                                    operand->getOperandNumber(),
-                                    operand->get()->getType().getAddressType());
+    return projectionBuilder.createTupleElementAddr(
+        pass.genLoc(), tupleAddr, operand->getOperandNumber(),
+        operand->get()->getType().getAddressType());
+  }
+  }
+}
+
+SILInstruction *
+AddressLoweringState::getProjectionInsertionPoint(SILValue userValue) const {
+  if (!valueStorageMap.contains(userValue)) {
+    return userValue->getDefiningInsertionPoint();
+  }
+  if (!valueStorageMap.getStorage(userValue).isProjection()) {
+    return userValue->getDefiningInsertionPoint();
   }
+  SILInstruction *retval = nullptr;
+  for (auto *pair : valueStorageMap.getProjections(userValue)) {
+    auto const &storage = pair->storage;
+    if (storage.storageAddress) {
+      // There's already an address.  Projections should be inserted after it.
+      if (auto *arg = dyn_cast<SILArgument>(storage.storageAddress)) {
+        // The address is a function argument.  Inserting before the first
+        // instruction will insert after it.
+        retval = &arg->getParent()->front();
+        break;
+      }
+      // The address is an instruction.  To insert after it, insertion must be
+      // before the next instruction.
+      auto *instruction = storage.storageAddress->getDefiningInstruction();
+      retval = instruction->getNextInstruction();
+      break;
+    }
```

Comment:
This if-block does one simple task: return the next instruction after the storage address. It could be a helper on the storage address.

Reply:
Fixed.  (ValueBase::getNextInstruction()`)